packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -459,15 +459,26 @@ class CupertinoPageTransition extends StatefulWidget {
     bool allowSnapshotting,
     Widget? child,
   ) {
-    final CurvedAnimation animation = CurvedAnimation(
-      parent: secondaryAnimation,
-      curve: Curves.linearToEaseOut,
-      reverseCurve: Curves.easeInToLinear,
-    );
-    final Animation<Offset> delegatedPositionAnimation = animation.drive(_kMiddleLeftTween);
-    animation.dispose();
+    final bool? linearTransition = ModalRoute.of(context)?.popGestureInProgress;
```

Comment:
Would probably be a bit more semantically correct to reference `Navigator.of(context).userGestureInProgress`, and logically we care about the state of the Navigator rather than any one route between the two for this. ModalRoute looks at that value for popGestureInProgress anyway.

Reply:
+1. Also as a nit I would name this `isLinearTransition` and make it non-nullable.